lib/course/widgets/course_page.dart

```diff
@@ -83,29 +83,31 @@ class CourseList extends StatelessWidget {
   Widget build(BuildContext context) {
     return PaginatedSliverList<Course>(
       dataLoader: Provider.of<CourseBloc>(context).getCourses,
-      itemBuilder: (context, course, __) => StreamBuilder<CourseSeries>(
-        stream: Provider.of<CourseBloc>(context)
-            .getCourseSeries(course.courseSeriesId),
-        builder: (context, snapshot) {
-          if (!snapshot.hasData)
+      itemBuilder: (context, course, __) => Builder(
+        builder: (context) => StreamBuilder<CourseSeries>(
+          stream: Provider.of<CourseBloc>(context)
+              .getCourseSeries(course.courseSeriesId),
+          builder: (context, snapshot) {
+            if (!snapshot.hasData)
+              return ListTile(
+                title: Text(snapshot.hasError
+                    ? snapshot.error.toString()
+                    : HpiL11n.get(context, 'loading')),
+              );
+
             return ListTile(
-              title: Text(snapshot.hasError
-                  ? snapshot.error.toString()
-                  : HpiL11n.get(context, 'loading')),
```

Comment:
Same as above, I think `snapshot.error?.toString() ?? HpiL11n.get(...)` would be cleaner.